### ISSUE TITLE
Shuffle_Block to TFShuffle_Block

### ISSUE THREAD

**Issue by larrywal-express** [ultralytics/yolov5#9971]
I will like to implement the below code in tf.py. From Shuffle_Block to TFShuffle_Block. can someone please help me. Thanks

def channel_shuffle(x, groups):
    batchsize, num_channels, height, width = x.data.size()
    channels_per_group = num_channels // groups

    # reshape
    x = x.view(batchsize, groups,
               channels_per_group, height, width)

    x = torch.transpose(x, 1, 2).contiguous()

    # flatten
    x = x.view(batchsize, -1, height, width)

    return x

class Shuffle_Block(nn.Module):
    def __init__(self, inp, oup, stride):
        super(Shuffle_Block, self).__init__()

        if not (1 <= stride <= 3):
            raise ValueError('illegal stride value')
        self.stride = stride

        branch_features = oup // 2
        assert (self.stride != 1) or (inp == branch_features << 1)

        if self.stride > 1:
            self.branch1 = nn.Sequential(
                self.depthwise_conv(inp, inp, kernel_size=3, stride=self.stride, padding=1),
                nn.BatchNorm2d(inp),
                nn.Conv2d(inp, branch_features, kernel_size=1, stride=1, padding=0, bias=False),
                nn.BatchNorm2d(branch_features),
                nn.ReLU(inplace=True),
            )

        self.branch2 = nn.Sequential(
            nn.Conv2d(inp if (self.stride > 1) else branch_features,
                      branch_features, kernel_size=1, stride=1, padding=0, bias=False),
            nn.BatchNorm2d(branch_features),
            nn.ReLU(inplace=True),
            self.depthwise_conv(branch_features, branch_features, kernel_size=3, stride=self.stride, padding=1),
            nn.BatchNorm2d(branch_features),
            nn.Conv2d(branch_features, branch_features, kernel_size=1, stride=1, padding=0, bias=False),
            nn.BatchNorm2d(branch_features),
            nn.ReLU(inplace=True),
        )

    @staticmethod
    def depthwise_conv(i, o, kernel_size, stride=1, padding=0, bias=False):
        return nn.Conv2d(i, o, kernel_size, stride, padding, bias=bias, groups=i)

    def forward(self, x):
        if self.stride == 1:
            x1, x2 = x.chunk(2, dim=1)
            out = torch.cat((x1, self.branch2(x2)), dim=1)
        else:
            out = torch.cat((self.branch1(x), self.branch2(x)), dim=1)

        out = channel_shuffle(out, 2)

        return out

**Comment by larrywal-express** [comment]
The code is from https://github.com/ppogg/YOLOv5-Lite/blob/master/models/common.py

**Comment by glenn-jocher** [comment]
@larrywal-express hi there! We'd be glad to assist you with the implementation. You can adapt the existing `Shuffle_Block` into `TFShuffle_Block` using TensorFlow. You'll need to replace the PyTorch operations with their equivalent in TensorFlow, such as using `tf.nn.depthwise_conv2d` for depthwise convolution and `tf.concat` for concatenation.

You'll also need to create a `channel_shuffle` function using TensorFlow operations for reshaping, transposing, and flattening the input tensor. 
If you encounter any issues during the implementation, feel free to ask for further assistance. Good luck with the conversion!
